packages/color-mode/src/color-mode-provider.tsx

```diff
@@ -57,109 +53,73 @@ export function ColorModeProvider(props: ColorModeProviderProps) {
   const {
     value,
     children,
-    options: { useSystemColorMode, initialColorMode },
+    options: { useSystemColorMode, initialColorMode } = {},
     colorModeManager = localStorageManager,
   } = props
 
   const defaultColorMode = initialColorMode === "dark" ? "dark" : "light"
 
-  /**
-   * Only attempt to retrieve if we're on the server. Else this will result
-   * in a hydration mismatch warning and partially invalid visuals.
-   *
-   * Else fallback safely to `theme.config.initialColormode` (default light)
-   */
   const [colorMode, rawSetColorMode] = React.useState<ColorMode | undefined>(
-    colorModeManager.type === "cookie"
-      ? colorModeManager.get(defaultColorMode)
-      : defaultColorMode,
+    defaultColorMode,
   )
 
   const { document } = useEnvironment()
 
   React.useEffect(() => {
-    /**
-     * Since we cannot initially retrieve localStorage for the reasons mentioned
-     * above, do so after hydration.
-     *
-     * Priority:
-     * - if `useSystemColorMode` is true system-color will be used as default - initial
-     * colormode is the fallback if system color mode isn't resolved
-     *
-     * - if `--chakra-ui-color-mode` is defined through e.g. `ColorModeScript` this
-     * will be used
-     *
-     * - if `colorModeManager` = `localStorage` and a value is defined for
-     * `chakra-ui-color-mode` this will be used
-     *
-     * - if `initialColorMode` = `system` system-color will be used as default -
-     * initial colormode is the fallback if system color mode isn't resolved
-     *
-     * - if `initialColorMode` = `'light'|'dark'` the corresponding value will be used
-     */
-    if (isBrowser && colorModeManager.type === "localStorage") {
-      const systemColorWithFallback = getColorScheme(defaultColorMode)
-      if (useSystemColorMode) {
-        return rawSetColorMode(systemColorWithFallback)
-      }
-      const rootGet = root.get()
-      const colorManagerGet = colorModeManager.get()
-      if (rootGet) {
-        return rawSetColorMode(rootGet)
-      }
-      if (colorManagerGet) {
-        return rawSetColorMode(colorManagerGet)
-      }
-      if (initialColorMode === "system") {
-        return rawSetColorMode(systemColorWithFallback)
-      }
-      return rawSetColorMode(defaultColorMode)
+    if (!isBrowser) return
+    const utils = getColorModeUtils({ doc: document })
+
+    const managerValue = colorModeManager.get()
+    if (managerValue) {
+      return rawSetColorMode(managerValue)
```

Comment:
Because `useEffect` treats a returned function as cleanup fn, this line gave me the impression `rawSetColorMode` returned a function, but is simply a setState call.

```suggestion
     rawSetColorMode(managerValue)
     return
```